docs/configuration.md

```diff
@@ -243,6 +243,45 @@ values, but here they are if you need to:
 ### regex
 This is the regex which is used to match the current branch to the correct branch configuration.
 
+### source-branches
+Because git is a directed graph GitVersion sometimes cannot tell which branch the current branch was branched from.
+
+Take this graph
+
+```
+  *   feature/foo
+* |   release/1.0.0
+|/
+*
+*
+* (master)
```

Comment:
It's not obvious to me what each branch label refers to in this diagram. I think the branch labels should really be shown next to the head commit of the branch.

Specifically, I'm not sure if you mean this:

```
             *   feature/foo
* master     |   release/1.0.0      // head of release branch is behind head of feature branch
| ___________/
|/
|
|
```

or this:

```
             *   feature/foo, release/1.0.0      // head of both branches is the same
* master     |
| ___________/
|/
|
|
|
```